accepted/single-file/staging.md

```diff
@@ -0,0 +1,139 @@
+# Single-file Staging 
+
+Publishing apps as a single file is a popular feature-request in .Net Core. Ideally, we want a single-file solution that:
+
+* Is compatible with all .Net Core applications
+* Bundles MSIL, R2R, native code and custom (data) files
+* Doesn't require installation or cleanup steps
+* Runs directly from the bundle, without extracting components to disk
+* Reduces publish-size
+* Improves startup cost
+* Works cohesively with debuggers, profilers, Watson dump etc. 
+
+This document explores a few options to realize the single-file publish feature, with different feature-set vs development cost trade-offs. The document can also be considered a development staging plan, where each stage spills fewer items onto temporary files, while paying an incremental development cost.
+
+## 1. Self-Extractor
+
+The first stage is to develop a pack and extract tool.
+While this stage is technically simplistic, the interface and tooling will match the final solution, giving partner teams and potential customers a chance to prepare for adopting the technology.
+
+### 1.1 Description
+
+This stage implements:
+
+* A  packaging tool (bundler) that embeds the application and all of its dependencies (essentially the contents of the publish directory) into a host executable. 
+	* This version will not support compression of the bundled assemblies.
+* When the executable runs, it will extract all of those files into a temporary directory, and then run as though the app was published to that temporary directory.
+
+The mechanism will support:
+
+* Reuse: Extract on first-run, reuse on subsequent runs.
+* Upgrade: Each version of the app extracts to a unique location, supporting side-by-side use of multiple versions.
+* Uninstall: Users can identify and delete extracted files when the app is no longer needed.
+* Access control: Processes running with elevated access can extract to admin-only-writable locations.
+
+### 1.2 Scenarios
+
+Best suited for:
+
+* Environments requiring maximal compatibility -- need to embed different kinds of files (IL, native, etc) into one, without losing functionality such as debuggability.
+
+Limitations:
+
+* Unsuitable for environments that require that the app does not perform disk-writes at startup
+
+Advantages:
+
+* Low cost of development
+* Bundler tool can be used as-is for most further stages
+* Provides ability to develop test infrastructure and prototypes
+
+## 2. Run from Bundle: MSIL
+
+### 2.1 Description
+
+This stage improves on Stage 1 in that 
+
+* MSIL files bundled into the single-file will load and execute directly from the executable.
+* Native libraries will still need to 
+	* Remain in the publish directory unmerged, or
+	* Extracted to the disk like the previous self-extractor stage
```

Comment:
What about an option that combines these two approaches? I.e. extract the native libraries to the current folder on first launch (and whenever they get deleted).

Reply:
This option can be achieved by using persistent-extraction, and setting the extraction-location as the current directory, using the configuration mentioned [here](https://github.com/dotnet/designs/pull/52/files#diff-5e93215a94e626495c10448a40fa4ccbR62).